### ISSUE TITLE
Discussion regarding adding a backend to doante animals section

### ISSUE THREAD

**Issue by akshitagupta15june** [akshitagupta15june/PetMe#346]
https://akshitagupta15june.github.io/PetMe/donate.html

Discuss in comments how can we proceed?

**Comment by Hayat4144** [comment]
Hey, [akshitagupta15june](https://github.com/akshitagupta15june). I checked it the link you provided and noticed  application used third party form submission . we should handle by us in backend with node js and Mongodb for storing the data with strict validation .
validation has been done but in the client side which is not so safe . we should validate all the user data in backend to prevent cross site scripting attack and other security issue like disk buffer overflow. I will add most of the features according to your requirement. if you want some other features i will try my best to add all the features you want.

**Comment by Rihem-taboubi** [comment]
I agree on adding a backend section to the donate animal form, we can use some technologies such as node js or PHP  and connect it with a database management system, Mongo DB   or SQL.and then we will handle the data easily.

**Comment by Cipher-08** [comment]
Adding a backend will allow to handle the data submitted by users, store it, and perform any necessary operations or integrations.
Server-side technology we should use to handle the backend. Options include Node.js .
Choices for databases  include **MySQL** or we can use **firebase** too as it will be easy to handle by a newbie.
Then the next step would be to **Update the HTML form in your page to include the necessary input fields for users to provide information about the donated animals.**
We can connect our backend application to the database and define the **required database schemas or models to represent the data too**!!!
We should also work on **implementing email notifications or confirmation messages** to inform users that their donation has been received successfully. This can be easily achieved by integrating with an email service provider or using a library specific to your chosen server-side technology.

**Comment by SHAIKASIFALI** [comment]
YES, I think we need to implement the entire backend for this service ..I can work on this issue under GSSOC 2023 kindly allot this issue to me

**Comment by suhaas02** [comment]
we can define a schema for storing the field in donate animals section and do data validation on backend using npm packages and render the details on the home page where we can show the available pets to adopt. @akshitagupta15june please assign this issue to me, i can work on this issue and i have complete idea on how to implement this feature.

**Comment by rafiya2003** [comment]
can you please assign me this issue under GSsoc'23

**Comment by KamalSritha-04** [comment]
can you please assign me this issue under GSSOC 23

**Comment by Pranavcm-07** [comment]
Hey @akshitagupta15june, I checked the link you provided. We can implement the backend with NodeJS and Mongo DB.
It seems like this issue is not yet assigned. I would like to work on this issue.
Can you assign me this issue?

**Comment by pnkjbee2** [comment]
hey, @akshitagupta15june , if this issue is still undone? i can work on it, i'll use node and express and postgres or mongo for backend ; i'll create a user class and put the relevant fields in it.

**Comment by VISHESH0932** [comment]
Hey @akshitagupta15june, I checked the link you provided. implementing  the backend with NodeJS and Mongo DB will be a good approach.
It seems like this issue is not yet assigned. I would like to work on this issue.
Can you assign me this issue?

**Comment by lovelymahor** [comment]
I would like to work on this project. Please assign it to me!
I checked the link you provided. We can implement the backend with Node.js or PHP and connect it with a database management system, such as MongoDB or SQL.

**Comment by Hmorel8263** [comment]
Hello! Just wanted to know if I can possibly take a crack at this issue!
